staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -43,40 +43,71 @@ import (
 	"k8s.io/kubectl/pkg/util/templates"
 )
 
-// AnnotateOptions have the data required to perform the annotate operation
-type AnnotateOptions struct {
-	PrintFlags *genericclioptions.PrintFlags
-	PrintObj   printers.ResourcePrinterFunc
+// AnnotateFlags directly reflect the information that CLI is gathering via flags.  They will be converted to Options, which
+// reflect the runtime requirements for the command.  This structure reduces the transformation to wiring and makes
+// the logic itself easy to unit test
+type AnnotateFlags struct {
+	// Common user flags
+	All           bool
+	AllNamespaces bool
 
-	// Filename options
+	DryRunStrategy cmdutil.DryRunStrategy
```

Comment:
What is the reason of these two DryRun fields(strategy and verifier) are in AnnotateFlags?

Reply:
That is very good question. I didn't know better. Do you think it makes more sense to leave them in AnnotateOptions? @ardaguclu

Reply:
I think, this(and DryRunVerifier) can be only in options.

Reply:
Yeah, try to stick with the rule, if it's not a flag then most likely it should go to `*Options` struct.